### PR TITLE
Pelias: Handle Custom Geocoder Place Types

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
This PR makes use of the responses generated by https://github.com/ibi-group/pelias-stitch/pull/1

Highlights stations and stops. Also adds https://github.com/mswjs/msw to test the input more effectively. 

Screen Shot 2021-08-31 at 2 49 23 PM

### CODE REVIEW

**Comment by miles-grant-ibigroup** [comment]
Merge blocked again waiting for #285, but ready for review

File: packages/location-field/src/styled.js [hunk 1]
```diff
@@ -138,9 +139,22 @@ export class MenuItem extends Component {
...
+      children,
+      color,
+      disabled,
+      header
+    } = this.props;
     return header ? (
-      <MenuItemHeader centeredText={centeredText}>{children}</MenuItemHeader>
+      <MenuItemHeader
+        className="header"
+        bgColor={color}
```
Comment (binh-dam-ibigroup) [inline packages/location-field/src/styled.js:153]: I would spell out `backgroundColor`.

File: packages/location-field/src/styled.js [hunk 2]
```diff
@@ -138,9 +139,22 @@ export class MenuItem extends Component {
...
+      centeredText,
+      children,
+      color,
+      disabled,
+      header
+    } = this.props;
     return header ? (
-      <MenuItemHeader centeredText={centeredText}>{children}</MenuItemHeader>
+      <MenuItemHeader
+        className="header"
```
Comment (binh-dam-ibigroup) [inline packages/location-field/src/styled.js:152]: Is this classname set in CSS anywhere (including OTP-RR)? If not, you could convert the header into a styled component and specify the hover effect in there.

File: packages/location-field/src/stories/Mobile.story.js [hunk 3]
```diff
@@ -129,6 +130,7 @@ export const Styled = () => (
   <StyledLocationField
     currentPosition={currentPosition}
     geocoderConfig={geocoderConfig}
+    layerColorMap={layerColorMap}
```
Comment (binh-dam-ibigroup) [inline packages/location-field/src/stories/Mobile.story.js:133]: Colors don't seem to take effect in any mobile stories, is that intended? (and if not, then why is the passing of the prop needed?)

File: packages/location-field/src/stories/Desktop.story.js [hunk 4]
```diff
@@ -104,3 +108,30 @@ export const WithBadApiKeyHandlesBadAutocomplete = () => (
...
+);
+
+export const WithPrefilledSearch = () => (
+  <LocationField
+    initialSearchResults={mockedGeocoderResponse.features}
+    layerColorMap={layerColorMap}
+    currentPosition={currentPosition}
+    geocoderConfig={geocoderConfig}
+    getCurrentPosition={getCurrentPosition}
+    inputPlaceholder="Select from location"
```
Comment (binh-dam-ibigroup) [inline packages/location-field/src/stories/Desktop.story.js:132]: Sort props.

File: packages/location-field/src/stories/Desktop.story.js [hunk 5]
```diff
@@ -1,14 +1,18 @@
...
+import { Ship } from "styled-icons/fa-solid";
 import LocationField from "..";
 import {
   currentPosition,
   geocoderConfig,
   getCurrentPosition,
   onLocationSelected,
-  selectedLocation
+  selectedLocation,
+  layerColorMap
```
Comment (binh-dam-ibigroup) [inline packages/location-field/src/stories/Desktop.story.js:11]: Sort names in this import.

File: packages/location-field/src/options.js [hunk 6]
```diff
@@ -31,6 +57,8 @@ export function Option({ disabled, icon, isActive, onClick, title }) {
 }
 
 Option.propTypes = {
+  classes: PropTypes.string,
```
Comment (binh-dam-ibigroup) [inline packages/location-field/src/options.js:60]: Add JSDoc to this prop.

File: packages/location-field/src/index.js [hunk 7]
```diff
@@ -356,38 +418,51 @@ class LocationField extends Component {
...
+        stationFeatures.map(feature =>
+          this.renderFeature(itemIndex++, feature)
+        ),
+
+        stopFeatures.length > 0 && (
+          <Styled.MenuItem
+            color={layerColorMap.stops}
+            centeredText
+            header
+            key="gtfs-stops-header"
```
Comment (binh-dam-ibigroup) [inline packages/location-field/src/index.js:453]: Sort props

File: packages/location-field/src/index.js [hunk 8]
```diff
@@ -356,38 +418,51 @@ class LocationField extends Component {
...
-          );
-          itemIndex++;
-          return option;
-        })
+        stationFeatures.length > 0 && (
+          <Styled.MenuItem
+            color={layerColorMap.stations}
+            centeredText
+            header
+            key="gtfs-stations-header"
```
Comment (binh-dam-ibigroup) [inline packages/location-field/src/index.js:439]: Sort props

File: packages/location-field/src/index.js [hunk 9]
```diff
@@ -356,38 +418,51 @@ class LocationField extends Component {
       // Add the menu sub-heading (not a selectable item)
       // menuItems.push(<MenuItem header key='sr-header'>Search Results</MenuItem>)
 
+      // Split out transit results
```
Comment (binh-dam-ibigroup) [inline packages/location-field/src/index.js:421]: Also add comment about keeping the first three results.

Comment (binh-dam-ibigroup) [reply packages/location-field/src/index.js:421]: Should `3` become a prop or something?

File: packages/location-field/src/index.js [hunk 10]
```diff
@@ -308,17 +318,71 @@ class LocationField extends Component {
...
+    // Create and return the option menu item
+    return (
+      <Option
+        icon={operatorIcon || <GeocodedOptionIconComponent feature={feature} />}
+        key={optionKey++}
+        title={feature.properties.label}
+        onClick={locationSelected}
+        isActive={itemIndex === activeIndex}
+        classes={classNames.join(" ")}
+        color={layerColorMap[layer]}
```
Comment (binh-dam-ibigroup) [inline packages/location-field/src/index.js:372]: Sort props

File: packages/location-field/src/index.js [hunk 11]
```diff
@@ -308,17 +318,71 @@ class LocationField extends Component {
...
   }
 
-  render() {
+  renderFeature = (itemIndex, feature) => {
     const {
+      layerColorMap,
+      geocoderConfig,
       addLocationSearch,
+      GeocodedOptionIconComponent,
+      operatorIconMap
```
Comment (binh-dam-ibigroup) [inline packages/location-field/src/index.js:327]: Sort vars

File: packages/location-field/src/styled.js (hunk 1 above)
Comment (binh-dam-ibigroup) [reply packages/location-field/src/styled.js:153]: Do we want to restrict the customization to the background color?

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Some sorting of props and vars, and please advise regarding applying the example color scheme to the mobile version.

File: packages/location-field/src/styled.js (hunk 1 above)
Comment (miles-grant-ibigroup) [reply packages/location-field/src/styled.js:153]: I would also prefer `backgroundColor`, but React throws errors when we do that. Have added a comment.

File: packages/location-field/src/styled.js (hunk 2 above)
Comment (miles-grant-ibigroup) [reply packages/location-field/src/styled.js:152]: Yes from what I can tell there are rules relating to `.header` menu times in OTP-RR

File: packages/location-field/src/stories/Mobile.story.js (hunk 3 above)
Comment (miles-grant-ibigroup) [reply packages/location-field/src/stories/Mobile.story.js:133]: I'm seeing colors in the mobile stories (when making a search). Are these not appearing for you?

File: packages/location-field/src/styled.js (hunk 1 above)
Comment (miles-grant-ibigroup) [reply packages/location-field/src/styled.js]: Customization support added in 53460ff1daa5fa52ad45e5f503596ed50f31b8ec

File: packages/location-field/src/stories/Desktop.story.js (hunk 4 above)
Comment (miles-grant-ibigroup) [reply packages/location-field/src/stories/Desktop.story.js]: Sorted in e0fc8a811358ed8a55a587f948db1614da68dbd4

File: packages/location-field/src/stories/Desktop.story.js (hunk 5 above)
Comment (miles-grant-ibigroup) [reply packages/location-field/src/stories/Desktop.story.js]: Sorted in e0fc8a811358ed8a55a587f948db1614da68dbd4

File: packages/location-field/src/options.js (hunk 6 above)
Comment (miles-grant-ibigroup) [reply packages/location-field/src/options.js:63]: Added in e0fc8a811358ed8a55a587f948db1614da68dbd4

File: packages/location-field/src/index.js (hunk 7 above)
Comment (miles-grant-ibigroup) [reply packages/location-field/src/index.js]: Sorted in e0fc8a811358ed8a55a587f948db1614da68dbd4

File: packages/location-field/src/index.js (hunk 8 above)
Comment (miles-grant-ibigroup) [reply packages/location-field/src/index.js]: Sorted in e0fc8a811358ed8a55a587f948db1614da68dbd4

File: packages/location-field/src/index.js (hunk 9 above)
Comment (miles-grant-ibigroup) [reply packages/location-field/src/index.js]: Good call! Comment and prop added in d030318b92928943de11c52377826231820d812d

File: packages/location-field/src/index.js (hunk 10 above)
Comment (miles-grant-ibigroup) [reply packages/location-field/src/index.js]: Sorted in e0fc8a811358ed8a55a587f948db1614da68dbd4

File: packages/location-field/src/index.js (hunk 11 above)
Comment (miles-grant-ibigroup) [reply packages/location-field/src/index.js]: Sorted in e0fc8a811358ed8a55a587f948db1614da68dbd4

**Comment by miles-grant-ibigroup** [comment]
Thanks for the feedback Binh! Good call with the enhanced customization.

File: packages/location-field/src/stories/Mobile.story.js (hunk 3 above)
Comment (binh-dam-ibigroup) [reply packages/location-field/src/stories/Mobile.story.js:134]: Okay, it's for when making a search.

**Review by binh-dam-ibigroup — APPROVED** [review]
Looks good

File: packages/location-field/src/index.js [hunk 12]
```diff
@@ -385,16 +385,17 @@ class LocationField extends Component {
...
+      nearbyStops,
       sessionOptionIcon,
       showClearButton,
       showUserSettings,
       static: isStatic,
       stopOptionIcon,
       stopsIndex,
       suppressNearby,
-      userLocationsAndRecentPlaces,
+      transitResultCount,
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply packages/location-field/src/index.js]: good call! renamed in 022dc2fb636cbb0ff44179e4fc971d422829ee8c

**Review by philip-cline — APPROVED** [review]
Looks good to me!

**Pull request merged**